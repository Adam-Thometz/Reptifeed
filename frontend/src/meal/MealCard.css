.MealCard {
  color: #D5CE3B;
  padding: 10px;
  width: 10vw;
  border-radius: 10px;
  margin: 0 auto;
}

.MealCard-img-wrapper {
  background-color: white;
  border-radius: 10px;
  padding: 10px;
  width: 85%;
}

.fruit, .supplement {
  color: #3E3E07;
}