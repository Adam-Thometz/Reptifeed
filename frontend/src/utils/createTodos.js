
/** createTodos(reptiles, pantry)
 * 
 * Takes a user's reptiles and pantry and creates a list of things they need to do to unlock the app's full features
 * 
 * Returns an two arrays of Link components, one of important things and one of nice-to-haves
 */

function createTodos(reptiles, pantry) {
  /** Essential checks:
   * - reptileCheck: Checks if a user has reptiles
   * - vegetableOftenCheck: Checks if a user has a vegetable staple in their pantry
   * - proteinOftenCheck: Checks if a user has a protein staple in their pantry
   * - treatModeratelyCheck: Checks if a user has a treat staple in their pantry
   * - supplementCheck: Checks if a user has at least two different supplements
   */
  const essentialTodos = [];

  const reptileCheck = !!reptiles.length;
  const vegetableOftenCheck = pantry.some(food => food.type === 'vegetable' && food.frequency === 'often');
  const proteinOftenCheck = pantry.some(food => food.type === 'protein' && food.frequency === 'often');
  const treatModeratelyCheck = pantry.some(food => food.isTreat && food.frequency === 'moderately');
  const supplementCheck = (pantry.filter(food => food.type === 'supplement')).length >= 2;

  if (!reptileCheck) essentialTodos.push('Add a reptile');
  if (!vegetableOftenCheck) essentialTodos.push('Add vegetables eaten often');
  if (!proteinOftenCheck) essentialTodos.push('Add protein eaten often');
  if (!treatModeratelyCheck) essentialTodos.push('Add a treat eaten moderately');
  if (!supplementCheck) essentialTodos.push('Add at least two supplements');

  /** Nice-to-have checks:
   * - vegetableModeratelyCheck: Checks if a user has a vegetable that's eaten moderately in their pantry
   * - vegetableOccasionallyCheck: Checks if a user has a vegetable that's eaten occasionally in their pantry
   * - proteinModeratelyCheck: Checks if a user has a protein that's eaten moderately in their pantry
   * - proteinOccasionallyCheck: Checks if a user has a protein that's eaten occasionally in their pantry
   * - treatOccasionallyCheck: Checks if a user has a treat that's eaten rarely in their pantry
   */
  const niceToHaveTodos = [];

  const vegetableModeratelyCheck = pantry.some(food => food.type === 'vegetable' && food.frequency === 'moderately');
  const vegetableOccasionallyCheck = pantry.some(food => food.type === 'vegetable' && food.frequency === 'occasionally' && !food.isTreat);
  const proteinModeratelyCheck = pantry.some(food => food.type === 'protein' && food.frequency === 'moderately');
  const proteinOccasionallyCheck = pantry.some(food => food.type === 'protein' && food.frequency === 'occasionally' && !food.isTreat);
  const treatOccasionallyCheck = pantry.some(food => food.isTreat && food.frequency === 'occasionally');

  if (!vegetableModeratelyCheck) niceToHaveTodos.push('Add a vegetable eaten moderately');
  if (!vegetableOccasionallyCheck) niceToHaveTodos.push('Add a vegetable eaten occasionally');
  if (!proteinModeratelyCheck) niceToHaveTodos.push('Add a protein eaten moderately');
  if (!proteinOccasionallyCheck) niceToHaveTodos.push('Add a protein eaten occasionally');
  if (!treatOccasionallyCheck) niceToHaveTodos.push('Add a treat eaten occasionally');

  return {essentialTodos, niceToHaveTodos};
};

export default createTodos;