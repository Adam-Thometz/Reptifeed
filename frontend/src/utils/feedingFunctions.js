/** Feeding functions
 * 
 * Used to get foods from pantry that match frequencies and food type.
*/

function getFoodFromPantry(pantry, freq, type) {
  const availableFood = pantry.filter(food => (!food.isTreat && (food.frequency === freq && food.type === type)));
  if (!availableFood.length) return `Get some more ${type}`
  const selectedIdx = Math.floor(Math.random() * availableFood.length);

  return availableFood[selectedIdx];
};

function getTreatFromPantry(pantry, freq) {
  const availableTreats = pantry.filter(food => food.isTreat && food.frequency === freq);
  if (!availableTreats.length) return "You don't have any treats";
  const selectedIdx = Math.floor(Math.random() * availableTreats.length); 
  
  return availableTreats[selectedIdx];
};

export { getFoodFromPantry, getTreatFromPantry };