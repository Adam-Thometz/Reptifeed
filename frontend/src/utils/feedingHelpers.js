/** Feeding functions
 * 
 * Used to get foods from pantry that match frequencies and food type.
 * 
 * If no foods are available for the frequency, returns foods eaten most often
*/

function getFoodFromPantry(pantry, freq, type) {
  let availableFood = pantry.filter(food => (!food.isTreat && (food.frequency === freq && food.type === type)));
  if (!availableFood.length) {
    availableFood = pantry.filter(food => (!food.isTreat && (food.frequency === 'often' && food.type === type)));
  };
  const selectedIdx = Math.floor(Math.random() * availableFood.length);

  return availableFood[selectedIdx];
};

function getTreatFromPantry(pantry, freq) {
  let availableTreats = pantry.filter(food => food.isTreat && food.frequency === freq);
  if (!availableTreats.length) {
    availableTreats = pantry.filter(food => food.isTreat);
  };
  const selectedIdx = Math.floor(Math.random() * availableTreats.length); 
  
  return availableTreats[selectedIdx];
};

function getNextSupplement(pantry, lastSupplement) {
  const nextSupplement = lastSupplement === 'calcium' ?
    pantry.find(food => food.name === 'multivitamin') :
    pantry.find(food => food.name === 'calcium');
  
  return nextSupplement;
}

export { getFoodFromPantry, getTreatFromPantry, getNextSupplement };