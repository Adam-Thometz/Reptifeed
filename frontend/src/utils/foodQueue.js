// foodQueue: Feed reptile based on frequency
const foodQueue = ["often", "often", "moderately", "often", "often", "moderately", "often", "often", "moderately", "occasionally"];

// treatQueue: Give reptile treats based on frequency
const treatQueue = ["moderately", "moderately", "occasionally"];

export { foodQueue, treatQueue };