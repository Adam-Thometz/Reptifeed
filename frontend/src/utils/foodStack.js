// foodStack: Feed reptile based on frequency
const foodStack = ["occasionally", "moderately", "often", "often", "moderately", "often", "often", "moderately", "often", "often"];

// treatStack: Give reptile treats based on frequency
const treatStack = ["occasionally", "moderately", "moderately"];

export { foodStack, treatStack };