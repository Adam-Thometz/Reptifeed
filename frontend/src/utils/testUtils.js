import React from "react";
import UserContext from "./UserContext";

const demoUser = {
  id: 1,
  username: 'rickSanchez',
  password: 'ilikeMorty',
  email: 'rickman@qmail.com',
  isAdmin: false
};

const demoReptile = [
  {
    id: 1,
    name: 'two crows',
    species: 'crows',
    subspecies: 'two of them',
    birthday: '2015-10-29',
    imgUrl: '',
    ownerId: 1
  }
];

const demoPantry = [
  {
    name: 'plumbus',
    type: 'everything',
    frequency: 'always',
    image: '',
    isTreat: true,
    tips: "Everyone needs a plumbus!"
  }
];

const UserProvider = ({ children, currUser = demoUser, pantry = demoPantry, reptiles = demoReptile }) => (
  <UserContext.Provider value={{currUser, pantry, reptiles}}>
    {children}
  </UserContext.Provider>
);

export { UserProvider };