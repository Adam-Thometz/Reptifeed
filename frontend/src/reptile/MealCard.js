import React from "react";
import './MealCard.css'

const MealCard = ({ name, type, image }) => {
  return (
    <div className={`MealCard ${type}`}>
      <div className="MealCard-img-wrapper">
        <img src={image} alt={name} />
      </div>
      <h5>{name}</h5>
    </div>
  );
};

export default MealCard;