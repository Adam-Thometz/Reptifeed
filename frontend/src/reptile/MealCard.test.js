import React from "react";
import MealCard from "./MealCard";
import { render, UserProvider, pantry } from '../utils/testUtils';
import { Routes, Route } from 'react-router-dom';

test('it renders without crashing', () => {
  const { name, type, image } = pantry[0];
  render(<UserProvider>
    <Routes>
      <Route path='/users/:id/reptiles/:reptileId' element={<MealCard name={name} type={type} image={image} />} />
    </Routes>
  </UserProvider>, { initialRoutes: ['/users/1/reptiles/1'] });
});

test('it matches snapshot', () => {
  const { name, type, image } = pantry[0];
  const { asFragment } = render(
    <UserProvider>
      <Routes>
        <Route path='/users/:id/reptiles/:reptileId' element={<MealCard name={name} type={type} image={image} />} />
      </Routes>
    </UserProvider>, { initialRoutes: ['/users/1/reptiles/1'] });
  expect(asFragment()).toMatchSnapshot();
});