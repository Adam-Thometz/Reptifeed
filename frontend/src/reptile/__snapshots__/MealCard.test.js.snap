// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MealCard everything"
  >
    <div
      class="MealCard-img-wrapper"
    >
      <img
        alt="plumbus"
        src=""
      />
    </div>
    <h5>
      plumbus
    </h5>
  </div>
</DocumentFragment>
`;
